src/%IPM/Lifecycle/Module.cls

```diff
@@ -69,11 +62,67 @@ Method %Activate(ByRef pParams) As %Status
 	Quit tSC
 }
 
+Method %PrepareDeploy(ByRef pParams) As %Status
+{
+	Set tSC = $$$OK
+	Try {
+		Set tVerbose = $Get(pParams("Verbose"))
+    Merge tParams = pParams
+
+		Set tNSConfig = ..Module.GetNamespaceConfig()
+
+		Set pParams("InitNamespace") = $Namespace
+
+		Set tNSPostfix = "PACKAGE-" _ $Piece($ZHorolog, ".")
+		Set tSC = ##class(%IPM.Utils.Build).InitializeNamespace(tNSPostfix, tVerbose, .tPackageDeployNamespace, tNSConfig)
+		$$$ThrowOnError(tSC)
+		Set pParams("PackageDeployNamespace") = tPackageDeployNamespace
+
+		$$$ThrowOnError(..Module.%Reload())
+
+		Set tSource = tPackageDeployNamespace _ "-CODE"
+
+		Set tKey = ""
+		For {
+			Set tResource = ..Module.Resources.GetNext(.tKey)
+			Quit:(tKey="")
+
+			Set tName = $Piece(tResource.Name, ".", 1, *-1)
+			Set tExtension = $$$ucase($Piece(tResource.Name, ".", *))
+			continue:tExtension=""
+
+			Set tSC = $Case(tExtension,
+				"GBL":##class(%IPM.Utils.Module).AddGlobalMapping(tPackageDeployNamespace, tName, tSource),
+				"INC":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "INC", tSource),
+				"INT":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "INT", tSource),
+				"MAC":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "MAC", tSource),
+				"PKG":##class(%IPM.Utils.Module).AddPackageMapping(tPackageDeployNamespace, tName, tSource),
+				"":##class(%IPM.Utils.Module).AddRoutineMapping(tPackageDeployNamespace, tName, "ALL", tSource))
```

Comment:
@isc-kiyer  Yeah definitely won't work. And I don't think we can add a single class as a package mapping (or can we?). I'm thinking about exportAsStream + importStream for .CLS resources into that namespace but not sure if that satisfies the need. Any thought?

Reply:
@Beiming-Zhang Copying classes seems like a waste of time as well though. Instead my suggestion would be to backup the database and restore if it the command fails rather than creating a separate namespace. If you want to copy classes, may as well copy everything like verify and just install the module in the new namespace and do the deploy there (this would probably be the cleanest option actually instead of backing up the database)

Reply:
@isc-kiyer  The last option won't be available until DP provides jira: https://usjira.iscinternal.com/browse/DP-415033 to install IPM easily with NS installation. I personally don't really like the idea of backing up the whole database for such a small operation. Copying only the .CLS resources does not seem to be too time consuming based on the number of .CLS resources in the largest module HSBASE. As of this point, I would like to implement the export+import stream solution as a temporary deal, and have a jira for replace that with module installation in the new NS. How do you like it?

(Actually module installation might not be easy as well even with DP-415033, since don't know where the module come from, whether itself/its dependencies are still in one of the repos, etc. so may still need to do a full copy somehow)